test/e2e/infinispan/credential_store_test.go

```diff
@@ -13,6 +13,36 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var config = `
+<infinispan
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="urn:infinispan:config:14.0 https://infinispan.org/schemas/infinispan-config-14.0.xsd
+                        urn:infinispan:server:14.0 https://infinispan.org/schemas/infinispan-server-14.0.xsd
+                        urn:org:jgroups http://www.jgroups.org/schema/jgroups-5.2.xsd"
+    xmlns="urn:infinispan:config:14.0"
+    xmlns:server="urn:infinispan:server:14.0">
+
+    <server xmlns="urn:infinispan:server:14.0">
+        <data-sources>
+          <data-source name="postgres" jndi-name="jdbc/postgres">
+            <connection-factory driver="org.postgresql.Driver"
+                                username="dbuser"
+                                url="${org.infinispan.server.test.postgres.jdbcUrl}">
```

Comment:
Isn't this going to cause issue given the test does not deploy database?

Reply:
Datasources configurations are validated on startup, but instances of the datasource are accessed lazily by caches. So we'll be fine as long as we don't create a Cache referencing the datasource in our test.

AFAICT all of the places in the server configuration that allow credential-store references are related to configuring connections to external resources, so this was the simplest way I could think to test without having to deploy anything else.